src/components/Search/utils.ts

```diff
@@ -127,16 +127,53 @@ export function getSearchQuery(
     nestedQuery,
     esPaginationOptions: {
       from: (Number(page) - 1 || 0) * (Number(offset) || 21),
-      size: Number(offset) || 21
+      size: Number(offset) >= 0 ? Number(offset) : 21
     },
-    sortOptions: { sortBy: sort, sortDirection },
+    sortOptions: sort ? { sortBy: sort, sortDirection } : undefined,
     filters
   } as BaseQueryParams
 
   const query = generateBaseQuery(baseQueryParams)
   return query
 }
 
+export interface AggregationQuery {
+  [x: string]: { terms: { field: string; size?: number } }
+}
+
+export interface AggregationResult {
+  [x: string]: { buckets: { key: string; doc_count: number }[] | number }
+}
+
+export const facetedQuery = (): AggregationQuery => {
+  const customFacetedSearch = process.env.NEXT_PUBLIC_FACETED_SEARCH_BUCKETS
+  const props = customFacetedSearch ? customFacetedSearch.split(';') : []
+  const terms = props.map((p) => p.split(','))
+  let customAgg: AggregationQuery = {}
+  terms.forEach((t) => {
+    customAgg = Object.assign(
+      JSON.parse(
+        `{ "${t[0]}": { "terms": { "field": "${t[1]}", "size": "${
+          t[2] ?? 100
+        }" } } }`
+      ),
+      customAgg
+    )
+  })
+  const basicAgg = {
+    tags: {
+      terms: { field: 'metadata.tags.keyword', size: 100 }
+    },
+    access: {
+      terms: { field: 'services.type.keyword', size: 100 }
+    },
+    service: {
```

Comment:
Are these keys correct? The access key uses 'services' as field value, whilst the service key uses metadata as field value

Reply:
Yes they are:

``` typescript
accessType && filters.push(getFilterTerm('services.type', accessType))
  serviceType && filters.push(getFilterTerm('metadata.type', serviceType))
```

Reply:
Obsolete, no hard-coded location anymore